### ISSUE TITLE
what do you think of training using forward hook mechanism instead of storing so many LoRA checkpoints

### ISSUE THREAD

**Issue by GenniusXu** [jerryliang24/Drag-and-Drop-LLMs#9]
Your training precedure requires pre-finetune the LLM and store a lot of LoRA checkpoints in the disk，and training the ParaGenerator to fitting the LoRA parameters. Why not use the ParaGenerator to generate a LoRA matrix，inject it in to the LLM using the forward hook mechanism，and calculate the cross entropy loss of the LLM’s output，then backward and update the ParaGenerator’s parameters directly？Do you think it is unstable to learn or has something drawbacks？

**Comment by jerryliang24** [comment]
Thanks for your interest in our work!

First, what you are discussing here can be described as an end-to-end parameter generation method, generate the tensor and directly use it as the weight to calculate the output. This is surely a valid way to carry out a series of interesting explorations, but it has nothing to do with the foward hook you metioned. In fact, I myself have tried a simple experiment on my own laptop like this:

<img width="1118" height="580" alt="Image" src="https://github.com/user-attachments/assets/476c7c26-bdc6-4230-894d-85327e49e06e" />


Feel free to make your own investigations in this direction and share your findings.
